pkg/apis/resource/validation/validation_resourceclaim_test.go

```diff
@@ -605,10 +610,206 @@ func TestValidateClaimStatusUpdate(t *testing.T) {
 				return claim
 			},
 		},
+		"valid-network-device-status": {
+			oldClaim: func() *resource.ResourceClaim { return validAllocatedClaim }(),
+			update: func(claim *resource.ResourceClaim) *resource.ResourceClaim {
+				claim = claim.DeepCopy()
```

Comment:
I think we're already good here, as when this test case gets invoked `update` will be called with a copy of `oldClaim` already. Is there a concern w/ function closure + parallelism race condition collisions?

In any event whichever pattern is provably correct w/ the least amount of excess we should apply to all test cases!

Reply:
I removed the DeepCopy, thank you. But I am not sure if I fully understood the other part of your comment.

Reply:
I was just wildly speculating on why it might make sense to make yet another copy. Not surprising that it didn't make sense!